tests/test_otel.py

```diff
@@ -257,8 +258,40 @@ def test_batches(client, trace_service, batch_count):
     assert len(trace_service.batches) == batch_count
 
     for batch in trace_service.batches:
-        assert get_attr(batch[0].resource, "service.name") == "test_service"
+        assert (
+            get_attr(batch[0].resource, "service.name")
+            == "unknown_service:nginx"
+        )
         assert len(batch[0].scope_spans[0].spans) == batch_size
 
     time.sleep(0.3)  # wait for +1 request to be flushed
     trace_service.batches.clear()
+
+
+@pytest.mark.parametrize(
+    "nginx_config",
+    [
+        {
+            "service_name": 'otel_service_name "test_service";',
```

Comment:
I don't think we need a separate template arg for "service_name", it's also part of a "resource". Also, multiline literal seems like a better fit here readability-wise, e.g.
```
            "resource": '''
                otel_service_name "test_service";
                otel_resource_attr my.name "my name";
                otel_resource_attr my.service "my service";
            '''
```

Reply:
Sure, changed.